File: searchIndex-platform/module/search-api/search-manager/app/controllers/SearchController.java
package controllers;

import org.ekstep.common.dto.Request;

import managers.PlaySearchManager;
import play.libs.F.Promise;
import play.mvc.Result;

/**
 * This is the main API that accepts the search request and returns the response. 
 * The API contract is identical to the graph data structures. 
 * So the returned data structures for each node type 
 * conform to the graph specification for that node type.
 */
public class SearchController extends SearchBaseController {

	private PlaySearchManager mgr = new PlaySearchManager();
	
	/**
	 * Method to search for all indexed data in the platform
	 * based on the request
	 * @return Promise<Result> containing searchResponse object
	 */
	public Promise<Result> search() {
		String apiId = "composite-search.search";
		Request request = getRequest(request().body(),apiId,request().uri());
		setHeaderContext(request(),request);
		Promise<Result> searchResponse = mgr.search(request);
		return searchResponse;
	}
	
	/**
	 * Method to get the count of indexed data in the platform 
	 * based on the request
	 * @return Promise<Result> containing count of data being searched
	 */
	public Promise<Result> count() {
		String apiId = "composite-search.count";
		Request request = getRequest(request().body(),apiId,request().uri());
		Promise<Result> response = mgr.count(request);
		return response;
	}

}

File: searchIndex-platform/module/search-api/search-manager/app/controllers/SearchController.scala
package controllers

import akka.actor.{ActorRef, ActorSystem}
import javax.inject.{Inject, Named}
import play.api.mvc.ControllerComponents

import scala.concurrent.ExecutionContext
import managers.PlaySearchManager

class SearchController@Inject()(@Named("SearchManager") actor: ActorRef, cc: ControllerComponents, actorSystem: ActorSystem)(implicit exec: ExecutionContext) extends BaseController(cc) {

    private val mgr = new PlaySearchManager

    /**
     * Method to search for all indexed data in the platform
     * based on the request
     *
     * @return Future[Result]
     */
    def search() = Action.async { implicit request =>
        val apiId: String = "composite-search.search"
        val internalReq = getRequest(apiId)
        setHeaderContext(internalReq)
        getResult(mgr.search(internalReq, actor))
    }

    /**
     * Method to get the count of indexed data in the platform
     * based on the request
     *
     * @return Future[Result]
     */
    def count() = Action.async { implicit request =>
        val apiId = "composite-search.count"
        val internalReq = getRequest(apiId)
        getResult(mgr.count(internalReq, actor))
    }
}
